### DISCUSSION TITLE
Sharing Auth State between main app and share extension

### DISCUSSION THREAD

**Discussion by pulkit-gsynergy** [invertase/react-native-firebase#4420] [Q&A]
How to share auth state between the main app and share extension in react-native?
I have added keychain groups in both of them with the same name and when I use auth().useUserAccessGroup(....), I am getting an error [Error: [auth/keychain-error] An error occurred when accessing the keychain. The @c NSLocalizedFailureReasonErrorKey field in the @c NSError.userInfo dictionary will contain more information about the error encountered]. 
Can someone help me with setting this up or help me with where I am going wrong.
How should the keychain group name be? I mean should it be <TEAM_ID>.<APP_NAME>.<GROUP_NAME>?

**Comment by mikehardy** [comment]
You may want to ask this upstream in firebase-ios-sdk, I don't believe react-native-firebase does anything related to the keychain? I think we are just wrapping the APIs

**Reply by pulkit-gsynergy** [reply]
react-native-firebase has a method "auth().useUserAccessGroup(....)" that makes use of a keychain group to get the auth state from another application.

**Reply by mikehardy** [reply]
@pulkit-gsynergy interesting - then you're already farther along than I am / would be in looking into this - I can only recommend digging into node_modules directly to see what the underlying code is doing and working through it that way

**Reply by pulkit-gsynergy** [reply]
@mikehardy could you help me with the ios code!
